verify-credentials.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+(function() {
+
+var logger = require('./setup').logger,
+    twitter = require('./setup').twitter;
+
+/**
+ * Ask Twitter to verify a user's credentials. If they are not valid,
+ * store the current time in user's deactivatedAt. If they are valid, clear
+ * the user's deactivatedAt. Save the user to DB if it's changed.
+ * A user can be deactivated because of suspension, deactivation, or revoked
+ * app. Each of these states (even revocation!) can be undone, and we'd
+ * like the app to resume working normally if that happens. So instead of
+ * deleting the user when we get one of these codes, store a 'deactivatedAt'
+ * timestamp on the user object. Users with a non-null deactivatedAt
+ * get their credentials retried once per day for 30 days, after which (TOD)
+ * they should be deleted. Regular operations like checking blocks or
+ * streaming are not performed for users with non-null deactivatedAt.
+ */
+function verifyCredentials(user) {
+  twitter.account('verify_credentials', {}, user.access_token,
+    user.access_token_secret, function(err, results) {
+      if (err && err.data) {
+        // For some reason the error data is given as a string, so we have to
+        // parse it.
+        var errJson = JSON.parse(err.data);
+        if (errJson.errors &&
+            errJson.errors.some(function(e) { return e.code === 89 })) {
+          logger.warn('User', user, 'revoked app.');
+          user.deactivatedAt = new Date();
+        } else if (err.statusCode === 404) {
+          //Testing for issue #146 indicates calling verify_credentials
```

Comment:
Re-reading the [docs](https://dev.twitter.com/rest/reference/get/account/verify_credentials), it looks like you get a 401 if credentials are not valid. Maybe add " || err.statusCode === 401" ?

You should be able to test the deactivation case pretty easily - go to the settings page and search for "deactivate my account" (tiny text at the bottom).